### PR TITLE
Remove composer.php / DATATABLES const

### PR DESCRIPTION
**Pull request by mvorisek**
`DATATABLES` const is kept in config files for security

### CODE REVIEW

**Comment by AllanJard** [comment]
Thanks for the PR. I'm not clear on why `if (!defined('DATATABLES')) exit();` is removed from the other files though? I wanted to ensure they couldn't be called directly from the web. I don't see they would do any damage, but just in case...

**Comment by mvorisek** [comment]
> why `if (!defined('DATATABLES')) exit();` is removed from the other files though? I wanted to ensure they couldn't be called directly from the web. I don't see they would do any damage, but just in case...

This is absolutely unneeded as including any class is not harmful as the class must be used to be hamful. And if included, the insecure place is not in the class, thus the check is fine to be removed.

**Comment by mvorisek** [comment]
@AllanJard can this PR be merged too? The global constant does not really provide any value nor harden security.

**Comment by mvorisek** [comment]
@AllanJard is the reasoning above enough, or are you still worried?

defined class (/wo any functional/global code) cannot execute itself in any way, if `if` statement is placed above a class, the file is still read/compiled, thus the security effect is exactly the same as before

**Review by AllanJard — COMMENTED** [review]
I'm not sure why `if (!defined('DATATABLES')) exit();` has been removed in most cases and reformatted in a few others? I'd rather it stayed in all cases.

**Comment by mvorisek** [comment]
> I'm not sure why `if (!defined('DATATABLES')) exit();` has been removed in most cases and reformatted in a few others? I'd rather it stayed in all cases.

In two places the `if (!defined('DATATABLES')) exit();` is kept to retain the original functionality. I reformatted them as curly braces are required by the recommended/PSR-12 coding standard.

In all other cases, the checks can/should be removed as they does not make any sense above classes. Loading a class is safe operation in sense of security as nothing is executed. Class itself should also not require any config. `composer.php` also does not have to be preloaded then and it allows native php classes preloading.

File: Bootstrap.php [hunk 1]
```diff
@@ -12,8 +12,11 @@
 
 
 namespace DataTables;
-if (!defined('DATATABLES')) exit();
 
+// Ensure Editor is loadable.
+if (!defined('DATATABLES') && !class_exists('\DataTables\Editor')) {
```
Comment (mvorisek) [inline Bootstrap.php:17]: @AllanJard I have updated this PR, now we check if `DATATABLES` const is defined - for BC and usage /wo composer.

With composer, the second check will do the work and no global `DATATABLES` const is even needed.

**Comment by AllanJard** [comment]
Good points - thank you!

**Comment by AllanJard** [comment]
I've restored the `DATATABLES` definition and reordered the Bootstrap file a little as the checks were causing it to exit out immediately.

**Pull request merged**